src/test/protocol/BuildInfo_test.cpp

```diff
@@ -1,106 +0,0 @@
-
-namespace ripple {
-
-class BuildInfo_test : public beast::unit_test::suite
-{
-public:
-    ProtocolVersion
-    from_version (std::uint16_t major, std::uint16_t minor)
-    {
-        return ProtocolVersion (major, minor);
-    }
-
-    void testValues ()
-    {
-        testcase ("comparison");
-
-        BEAST_EXPECT(from_version (1,2) == from_version (1,2));
-        BEAST_EXPECT(from_version (3,4) >= from_version (3,4));
-        BEAST_EXPECT(from_version (5,6) <= from_version (5,6));
-        BEAST_EXPECT(from_version (7,8) >  from_version (6,7));
-        BEAST_EXPECT(from_version (7,8) <  from_version (8,9));
-        BEAST_EXPECT(from_version (65535,0) <  from_version (65535,65535));
-        BEAST_EXPECT(from_version (65535,65535) >= from_version (65535,65535));
-    }
-
-    void testStringVersion ()
-    {
-        testcase ("string version");
-
-        for (std::uint16_t major = 0; major < 8; major++)
-        {
-            for (std::uint16_t minor = 0; minor < 8; minor++)
-            {
-                BEAST_EXPECT(to_string (from_version (major, minor)) ==
-                    std::to_string (major) + "." + std::to_string (minor));
-            }
-        }
-    }
-
-    void testVersionPacking ()
-    {
-        testcase ("version packing");
-
-        BEAST_EXPECT(to_packed (from_version (0, 0)) == 0);
-        BEAST_EXPECT(to_packed (from_version (0, 1)) == 1);
-        BEAST_EXPECT(to_packed (from_version (0, 255)) == 255);
-        BEAST_EXPECT(to_packed (from_version (0, 65535)) == 65535);
-
-        BEAST_EXPECT(to_packed (from_version (1, 0)) == 65536);
-        BEAST_EXPECT(to_packed (from_version (1, 1)) == 65537);
-        BEAST_EXPECT(to_packed (from_version (1, 255)) == 65791);
-        BEAST_EXPECT(to_packed (from_version (1, 65535)) == 131071);
-
-        BEAST_EXPECT(to_packed (from_version (255, 0)) == 16711680);
-        BEAST_EXPECT(to_packed (from_version (255, 1)) == 16711681);
-        BEAST_EXPECT(to_packed (from_version (255, 255)) == 16711935);
-        BEAST_EXPECT(to_packed (from_version (255, 65535)) == 16777215);
-
-        BEAST_EXPECT(to_packed (from_version (65535, 0)) == 4294901760);
-        BEAST_EXPECT(to_packed (from_version (65535, 1)) == 4294901761);
-        BEAST_EXPECT(to_packed (from_version (65535, 255)) == 4294902015);
-        BEAST_EXPECT(to_packed (from_version (65535, 65535)) == 4294967295);
-    }
-
-    void run () override
-    {
-        testValues ();
-        testStringVersion ();
-        testVersionPacking ();
-
-        auto const current_protocol = BuildInfo::getCurrentProtocol ();
-        auto const minimum_protocol = BuildInfo::getMinimumProtocol ();
-
-        BEAST_EXPECT(current_protocol >= minimum_protocol);
-
-        log <<
-            "   Ripple Version: " << BuildInfo::getVersionString() << '\n' <<
-            " Protocol Version: " << to_string (current_protocol) << std::endl;
-    }
-};
-
-BEAST_DEFINE_TESTSUITE(BuildInfo,ripple_data,ripple);
```

Comment:
As above, I am always reluctant to get rid of unit tests without a really good reason. I don't think you've entirely removed the functionality being tested here, so why remove the tests?

Reply:
This code changes `ProtocolVersion` to be a simple `std::pair` and all the relevant operators that were being tested here before are now provided by std so I don't believe that testing _that_ functionality is necessary.

The only test that is no longer there is BEAST_EXPECT(current_protocol >= minimum_protocol) and the new implementation of this code simply doesn't have such values in a way that makes sense to test. Rather the implementation requires that all supported versions are listed in a single array `supportedProtocolList`. I had a `static_assert` that was meant to test, but it turns out I had incorrectly gated it so that it would require compiling with C++20; I rewrote the `static_assert` to execute under C++17 and verified that it works correctly.